language/tools/move-mv-llvm-compiler/src/stackless/translate.rs

```diff
@@ -628,14 +645,28 @@ impl<'mm, 'up> FunctionContext<'mm, 'up> {
             }
         }
 
-        // Store params into locals
+        // Store params into locals.
+        //
+        // To support testing of scripts that require signers, inject signers that were provided
+        // on the command line into all script function arguments of type `signer`. Each `signer`
+        // argument is assigned in order from the command line signer list.
         {
             let param_count = self.env.get_parameter_count();
             let ll_params = (0..param_count).map(|i| ll_fn.get_param(i));
+            let is_script = self.env.is_script();
+            let mut curr_signer = 0;
 
             for (ll_param, local) in ll_params.zip(self.locals.iter()) {
-                self.llvm_builder
-                    .store_param_to_alloca(ll_param, local.llval);
+                if is_script && local.mty == mty::Type::Primitive(mty::PrimitiveType::Signer) {
+                    let signer = test_signers[curr_signer].strip_prefix("0x");
+                    curr_signer += 1;
+                    let addr_val = BigUint::parse_bytes(signer.unwrap().as_bytes(), 16);
+                    let c = self.constant(&sbc::Constant::Address(addr_val.unwrap()));
+                    self.llvm_builder.build_store(c.get0(), local.llval);
+                } else {
+                    self.llvm_builder
+                        .store_param_to_alloca(ll_param, local.llval);
+                }
```

Comment:
It's ok for testing, however, isn't signer supposed to be passed to the script by the VM via ABI?